### PR TITLE
Fix the JwtUtil.getUserId method bug

### PR DESCRIPTION
**Pull request by nuo-promise**
Fixes [https://github.com/dromara/shenyu/issues/1562](https://github.com/dromara/shenyu/issues/1562)

Make sure that:

- [x] You have read the [contribution guidelines](https://dromara.org/projects/soul/contributor/).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-admin/pom.xml [hunk 1]
```diff
@@ -233,6 +233,12 @@
             <artifactId>aspectjweaver</artifactId>
             <version>1.9.6</version>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/com.auth0/java-jwt -->
+        <dependency>
+            <groupId>com.auth0</groupId>
+            <artifactId>java-jwt</artifactId>
+            <version>3.8.3</version>
```
Comment (yu199195) [inline shenyu-admin/pom.xml:240]: this version define to root pom.xml

File: shenyu-admin/pom.xml [hunk 2]
```diff
@@ -233,6 +233,12 @@
...
+        </dependency>
```
Comment (dengliming) [inline shenyu-admin/pom.xml:241]: I think this dependency is duplicated.

**Comment by nuo-promise** [comment]
I'll see how to remove it

dengliming ***@***.***>于2021年6月2日 周三11:00写道：

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/shiro/config/ShiroRealm.java [hunk 3]
```diff
@@ -64,19 +71,32 @@ protected AuthorizationInfo doGetAuthorizationInfo(final PrincipalCollection pri
...
 
-        if (JwtUtils.getUserId() == null) {
-            JwtUtils.setUserId(token.getToken());
+    /**
+     * check token valid.
+     *
+     * @param token user token
+     * @return userInfo {@link UserInfo}
+     */
+    private UserInfo checkUserTokenIsValid(final String token) {
```
Comment (dengliming) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/shiro/config/ShiroRealm.java:90]: Does `getUserInfoByToken/getUserInfo` would be better? I suggest throws `AuthenticationException` directly If the user can't not be found.

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/shiro/config/ShiroRealm.java [hunk 4]
```diff
@@ -64,19 +71,32 @@ protected AuthorizationInfo doGetAuthorizationInfo(final PrincipalCollection pri
...
+        String userName = JwtUtils.getIssuer(token);
+        UserInfo userInfo = UserInfo.builder().build();
+        if (StringUtils.isNotEmpty(userName)) {
+            userInfo.setUserName(userName);
+            DashboardUserVO dashboardUserVO = dashboardUserService.findByUserName(userName);
+            if (dashboardUserVO != null) {
+                userInfo.setUserId(dashboardUserVO.getId());
+            }
         }
```
Comment (dengliming) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/shiro/config/ShiroRealm.java:99]:
```suggestion
        String userName = JwtUtils.getIssuer(token);
        if (StringUtils.isEmpty(userName)) {
            throw new AuthenticationException("userName is null");
        }
        DashboardUserVO dashboardUserVO = dashboardUserService.findByUserName(userName);
        if (dashboardUserVO == null) {
            throw new AuthenticationException(String.format("userName(%s) can not be found.", userName));
        }
        UserInfo userInfo = UserInfo.builder()
               .userName(userName)
               .userId(dashboardUserVO.getId())
               .build();        
```

**Review by dengliming — APPROVED** [review]

**Pull request merged**